sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/resource/AzureStorageFileProtocolResolver.java

```diff
@@ -70,7 +64,7 @@ public String getName() {
         @Override
         public Stream<StorageItem> listItems(String itemPrefix) {
             ShareClient shareClient = getShareServiceClient().getShareClient(name);
-            if (shareClient.exists()) {
+            if (Boolean.TRUE.equals(shareClient.exists())) {
```

Comment:
which sonar issue is this?

Reply:
Said Primitive type only can be used  directly. Yes, this makes sense and also other places we really used the recommended way.  I will paste the issue tracker here.

Reply:
refer: https://sonarcloud.io/project/issues?directories=sdk%2Fspring%2Fspring-cloud-azure-autoconfigure%2Fsrc%2Fmain%2Fjava%2Fcom%2Fazure%2Fspring%2Fcloud%2Fautoconfigure%2Faad%2Ffilter&resolved=false&types=CODE_SMELL&id=stliu_azure-sdk-for-java&open=AXtAG3Fk00FAjxsg1BZT

Reply:
I don't think this makes any sense

Reply:
will ignore this suggestion